### ISSUE TITLE
Build walls at more angles

### ISSUE THREAD

**Issue by Monsterfisch** [Sh0wdown/UnofficialCrusaderPatch#221]
Would it be possible to make it so that the player can build walls similar to the aiv editor at not just 90 and 45-degree angles but also at the 35 degrees or maybe even any angle?

might even be nice to go as far as have the wall flow around objects like trees and buildings (same as in the aiv editor) that way it would be a lot more convenient to actually build an effective and good looking castle as it would more reliably close in potential gaps in the wall.

here an example of the flowing and the different angles 

![](https://i.imgur.com/ACcKVav.png)

**Comment by KingCaliph** [comment]
I see the appeal of extra angles when it comes to designing castles. However who exactly would it work in game? I'm assuming people would not be able to work through any cracks, as usual. Not bashing the idea but I think there are bigger problems that go along with this suggestion, however I would love to see it be done. P.S. I see you around a lot on these forums MonsterFisch, glad to see a loyal fan

**Comment by Monsterfisch** [comment]
What kind of problems are you thinking of that could arise from this? 
By working through the cracks I assume you mean workers could pass between the wall segments?
If so that wouldn't be an issue as this is already not the case in the game as 45° diagonal walls already prevent that from happening.

**Comment by KingCaliph** [comment]
I should have been more clear. I meant when connecting wall. For instance, and I could be wrong about all of this, imagine you have a horizontal line and you draw a line at a 15 degree angle though it (the line is coming for the top left to the bottom right). There would be a point where a small portion of a 1x1 unit of wall sticking out of the horizontal line. The only reason a small part would stick out because each wall unit is set to a specific size. However, this problem could be fixed if the walls were instead forced together depending on how much was sticking out, or we could just ignore the "sticking out" problem in general and let the piece be. 
![Untitled drawing](https://user-images.githubusercontent.com/48774888/74280506-861aca80-4cd1-11ea-9460-0c52db10d2e5.png)

**Comment by Monsterfisch** [comment]
maybe this explains more what I am looking for as I feel we are still not on the same page ^^
![walls](https://i.imgur.com/KIzWenJ.jpg)

**Comment by KingCaliph** [comment]
Oh I see now, it makes a lot more sense now that I see what you are going for. In my mind I was picturing something different but the illustration shows how it would work. The walls would in a way mesh together leaving no gaps, as I was previously thinking. Well I'm convinced, having something like that would open up a lot more designs for castles, hopefully it can become a reality.

**Comment by Monsterfisch** [comment]
yeah especially since I made my aiv set with realistic castles ists been sorta bugging me that there is always that 45° angled placement of walls that for some reason is still a thing in the newer titles of the game except for sh3 which was a disaster. For other reasons though.

I mean imagine you could build something more like this without having to place down every wall piece manually how sick would that be =D https://i.imgur.com/yii8PeN.jpg

**Comment by KingCaliph** [comment]
It would be great, so many options with the drag of a mouse. It would add a lot to castles, appearance wise, and help bring the game to life. If all the AI had castles with interesting angles it would almost feel like a different game, considering how often you constantly see the 90 and 45 degree angled castles. Despite the fact that the angle of the walls may sound like a small detail, considering how the walls are constantly used by all AI it would be nice to have the option of more intricate walls.

**Comment by Monsterfisch** [comment]
100% agreed that's why I made my own set of 128 castles all of them are basically odd angles ^^

feel free to check it out here (has some aic to go along as well which I am still refining ofc)
[Strongholds_of_conquest](https://github.com/Monsterfisch/StrongholdsOfConquest_)

**Comment by KingCaliph** [comment]
128? wow that is a lot for just castles with angles.  I am definitely gonna check those out, thanks for sharing them.

**Comment by Monsterfisch** [comment]
Well it's not just angles but you will see ;)
